evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -1519,3 +1519,30 @@ def test_partial_dependence_not_allowed_types(types, cols, expected_cols):
         return
     s = partial_dependence(pl, X, cols, grid_resolution=2)
     assert not s.isnull().any().any()
+
+
+def test_partial_dependence_categorical_nan(fraud_100):
+    X, y = fraud_100
+    X.ww["provider"][:10] = None
+    pl = BinaryClassificationPipeline(
+        component_graph=[
+            "Imputer",
+            "DateTime Featurization Component",
+            "One Hot Encoder",
+            "Random Forest Classifier",
+        ]
+    )
+    pl.fit(X, y)
+
+    dep = partial_dependence(pl, X, features="provider", grid_resolution=5)
+
+    assert dep.shape[0] == X["provider"].dropna().nunique()
+    assert not dep["feature_values"].isna().any()
+    assert not dep["partial_dependence"].isna().any()
+
+    dep2way = partial_dependence(
+        pl, X, features=("amount", "provider"), grid_resolution=5
+    )
+    assert not dep2way.isna().any().any()
+    # Minus 1 in the columns because there is `class_label`
+    assert dep2way.shape == (5, X["provider"].dropna().nunique() + 1)
```

Comment:
Omega nitpick, but I think it'd be a good idea to set grid_resolution as a variable and use it above / here, where``assert dep2way.shape == (grid_resolution_variable,...)``? Just so its more clear where this 5 value is coming from :)

Reply:
I completely agree!